mfe/mfe.go

```diff
@@ -0,0 +1,1209 @@
+	}
+
+	for i := 0; (i * 5) < len(Triloops); i++ {
+		params.triloop[i] = rescaleDg(Triloop37[i], TriloopdH[i], tempf)
+	}
+
+	for i := 0; (i * 9) < len(Hexaloops); i++ {
+		params.hexaloop[i] = rescaleDg(Hexaloop37[i], HexaloopdH[i], tempf)
+	}
+
+	for i := 0; i <= nbPairs; i++ {
+		params.multiLoopIntern[i] = rescaleDg(ML_intern37, ML_interndH, tempf)
+	}
+
+	/* stacks    G(T) = H - [H - G(T0)]*T/T0 */
+	for i := 0; i <= nbPairs; i++ {
+		for j := 0; j <= nbPairs; j++ {
+			params.stackingPair[i][j] = rescaleDg(stack37[i][j],
+				stackdH[i][j],
+				tempf)
+		}
+	}
+
+	/* mismatches */
+	for i := 0; i <= nbPairs; i++ {
+		for j := 0; j < 5; j++ {
+			for k := 0; k < 5; k++ {
+				var mm int
+				params.mismatchInteriorLoop[i][j][k] = rescaleDg(mismatchI37[i][j][k],
+					mismatchIdH[i][j][k],
+					tempf)
+				params.mismatchHairpinLoop[i][j][k] = rescaleDg(mismatchH37[i][j][k],
+					mismatchHdH[i][j][k],
+					tempf)
+				params.mismatch1xnInteriorLoop[i][j][k] = rescaleDg(mismatch1nI37[i][j][k],
+					mismatch1nIdH[i][j][k],
+					tempf)
+				params.mismatch2x3InteriorLoop[i][j][k] = rescaleDg(mismatch23I37[i][j][k],
+					mismatch23IdH[i][j][k],
+					tempf)
+				// if model_details.dangles > 0 {
+				mm = rescaleDg(mismatchM37[i][j][k],
+					mismatchMdH[i][j][k],
+					tempf)
+				if mm > 0 {
+					params.mismatchMultiLoop[i][j][k] = 0
+				} else {
+					params.mismatchMultiLoop[i][j][k] = mm
+				}
+
+				mm = rescaleDg(mismatchExt37[i][j][k],
+					mismatchExtdH[i][j][k],
+					tempf)
+				if mm > 0 {
+					params.mismatchExteriorLoop[i][j][k] = 0
+				} else {
+					params.mismatchExteriorLoop[i][j][k] = mm
+				}
+				// } else {
+				// 	params.mismatchExt[i][j][k] = 0
+				// 	params.mismatchM[i][j][k] = 0
+				// }
+			}
+		}
+	}
+
+	/* dangles */
+	for i := 0; i <= nbPairs; i++ {
+		for j := 0; j < 5; j++ {
+			var dd int
+			dd = rescaleDg(dangle5_37[i][j],
+				dangle5_dH[i][j],
+				tempf)
+			if dd > 0 {
+				params.dangle5[i][j] = 0
+			} else {
+				params.dangle5[i][j] = dd
+			}
+
+			dd = rescaleDg(dangle3_37[i][j],
+				dangle3_dH[i][j],
+				tempf)
+			if dd > 0 {
+				params.dangle3[i][j] = 0
+			} else {
+				params.dangle3[i][j] = dd
+			}
+		}
+	}
+
+	/* interior 1x1 loops */
+	for i := 0; i <= nbPairs; i++ {
+		for j := 0; j <= nbPairs; j++ {
+			for k := 0; k < 5; k++ {
+				for l := 0; l < 5; l++ {
+					params.interior1x1Loop[i][j][k][l] = rescaleDg(int11_37[i][j][k][l],
+						int11_dH[i][j][k][l],
+						tempf)
+				}
+			}
+		}
+	}
+
+	/* interior 2x1 loops */
+	for i := 0; i <= nbPairs; i++ {
+		for j := 0; j <= nbPairs; j++ {
+			for k := 0; k < 5; k++ {
+				for l := 0; l < 5; l++ {
+					var m int
+					for m = 0; m < 5; m++ {
+						params.interior2x1Loop[i][j][k][l][m] = rescaleDg(int21_37[i][j][k][l][m],
+							int21_dH[i][j][k][l][m],
+							tempf)
+					}
+				}
+			}
+		}
+	}
+
+	/* interior 2x2 loops */
+	for i := 0; i <= nbPairs; i++ {
```

Comment:
Why so many nests?

Reply:
Not sure what we can do about this. We have to iterate through every value of the matrix to set it.